packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -240,6 +235,24 @@ export const supportPaste = browser.isNative || (!browser.isChrome && document.q
 
 export const RECENT_COMMANDS_STORAGE_KEY = 'commands';
 
+@injectable() export abstract class DomCommandHandler implements CommandHandler {
+    constructor(@unmanaged() protected domCommand: string) { }
+    execute(): void { document.execCommand(this.domCommand); }
+}
+@injectable() export abstract class NativeTextInputCommandHandler extends DomCommandHandler {
```

Comment:
`Native` looks misleading, I usually think about Electron

we also have `browser.isNative` which means something else

Maybe use `Dom` instead of `Native`?

Reply:
> Native looks misleading, I usually think about Electron

The naming is from VSCode: https://github.com/microsoft/vscode/blob/d9d476d17a454ffb84d384be0e81b45dfaa9e059/src/vs/editor/browser/controller/coreCommands.ts#L1597

I do not mind changing the name, but the current one is at least in sync with VS Code.

Reply:
If it end up being a bit confusing, you can still carefully explain the meaning with a jsdoc comment.

Reply:
I don't think we need to copy stuff from VS Code to our core as is. It does not even important for VS Code compatibility. To me is good enough that we learn something from there, but it is up to you.

Reply:
> I don't think we need to copy stuff from VS Code to our core as is

Reply:
> but it is up to you.

Then we leave as-is unless you have a good reason to change it.

Reply:
> I did not copy, why do you say I have copied something.

I meant naming, code. Sorry for the confusion.